code/game/machinery/lightswitch.dm

```diff
@@ -61,3 +65,23 @@
 		return
 	if(!(stat & (BROKEN|NOPOWER)))
 		power_change()
+
+/obj/machinery/light_switch/proc/shock(mob/user)
+	if(stat & (BROKEN|NOPOWER))		// unpowered, no shock
+		return FALSE
+	var/datum/effect_system/spark_spread/s = new /datum/effect_system/spark_spread
+	s.set_up(5, 1, src)
+	s.start()
+	if (electrocute_mob(user, get_area(src), src, 0.7, TRUE))
+		return TRUE
+	else
+		return FALSE
+
+/obj/machinery/light_switch/emag_act(mob/user)
+	if(obj_flags & EMAGGED)
+		to_chat(user, "<span class='warning'>Nothing new seems to happen when you swipe the emag.</span>")
+		return
+	to_chat(user, "<span class='notice'>You swipe the emag on the light switch. The light switch gives of a soft hum.</span>")
+	if(user.can_hear())
+		to_chat(user, "<span class='notice'>The light switch gives of a soft hum.</span>")
```

Comment:
The light switch gives **off** a soft hum.

Reply:
Committed.